presentation/components/src/presentation-components/table/DataProvider.ts

```diff
@@ -295,6 +390,9 @@ const createRow = (descriptor: Readonly<Descriptor>, item: Readonly<Item>): RowI
     // note: for table view we expect the record to always have only 1 primary key
     throw new PresentationError(PresentationStatus.InvalidArgument, "item.primaryKeys");
   }
+  const { firstExtractedFieldNameToNestedFieldMap: sameInstanceFieldsMap, updatedFields } = getFieldsWithExtractedSameInstanceFieldsAndCreateMap(descriptor.fields);

```

Comment:
We're doing this for every row, although the result is always the same. Could we do it once and share the result between all rows?

Reply:
Sure, updated the code so that now we would get `sameInstanceFieldsMap` and `updatedFields` in createRows function and pass them for every row from there.